core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java

```diff
@@ -133,7 +134,7 @@ public CommitService service(int rewritesPerCommit) {
   public class CommitService implements Closeable {
     private final ExecutorService committerService;
     private final ConcurrentLinkedQueue<RewriteFileGroup> completedRewrites;
-    private final List<RewriteFileGroup> committedRewrites;
+    private final List<RewriteDataFiles.FileGroupRewriteResult> committedRewrites;
```

Comment:
This should not expose the concrete class.

Reply:
I'm not sure if this modification is standard, because the `asResults()` method in the current `RewriteFileGroup` class should actually be written like this:  
https://github.com/apache/iceberg/blob/0f3e7c7badf47bd0e2178591082886cd47d94a1b/core/src/main/java/org/apache/iceberg/actions/RewriteFileGroup.java#L67-L70  
`  public RewriteDataFiles.FileGroupRewriteResult asResult() {  
    Preconditions.checkState(addedFiles != null, "Cannot get result, Group was never rewritten");  
    return new BaseFileGroupRewriteResult(info, addedFiles.size(), fileScanTasks.size());  
  }`